src/app/shared/services/dom-file-download.service.ts

```diff
@@ -9,4 +11,12 @@ export class DomFileDownloadService implements FileDownloadService {
     anchorElement.setAttribute('download', filename)
     anchorElement.click()
   }
+
+  public downloadJson(data: unknown, filename: string): void {
+    const text: string = JSON.stringify(data)
+    if (!filename.includes(JSON_EXTENSION)) {
```

Comment:
`.endsWith(JSON_EXTENSION)` would be more correct, since the file extension should be located at the end of the filename. Perhaps even ```.endsWith(`.${JSON_EXTENSION}`)```

Reply:
Updated